src/main/scala/dotty/dokka/ScalaPageCreator.scala

```diff
@@ -0,0 +1,91 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.base.translators.documentables.{DefaultPageCreator, PageContentBuilder, PageContentBuilder$DocumentableContentBuilder}
+import org.jetbrains.dokka.base.signatures.SignatureProvider
+import org.jetbrains.dokka.base.transformers.pages.comments.CommentsToContentConverter
+import org.jetbrains.dokka.transformers.documentation.DocumentableToPageTranslator
+import org.jetbrains.dokka.utilities.DokkaLogger
+import org.jetbrains.dokka.model._
+import org.jetbrains.dokka.pages._
+import collection.JavaConverters._
+import org.jetbrains.dokka.model.properties._
+
+
+class ScalaPageCreator(
+    commentsToContentConverter: CommentsToContentConverter,
+    signatureProvider: SignatureProvider,
+    val logger: DokkaLogger
+) extends DefaultPageCreator(commentsToContentConverter, signatureProvider, logger) {
+    override def pageForClasslike(c: DClasslike): ClasslikePageNode = {
+        val res = super.pageForClasslike(c)
```

Comment:
```
super.pageForClasslike(c) match
  case c: DClass => 
     clazz.get(ClasslikeExtension).fold(c){ co =>
            res.modified(
                    res.getName,
                    res.getContent,
                    res.getDri,
                    res.getEmbeddedResources,
                    (res.getChildren.asScala ++ List(
                        ClasslikePageNode(
                            co.getName + "$",
                            contentForClasslike(co),
                            Set(co.getDri).asJava,
                            co,
                            (co.getClasslikes.asScala.map(pageForClasslike(_)) ++
                            co.getFunctions.asScala.map(pageForFunction(_))).asJava,
                            Nil.asJava
                        )
                    )).asJava
                )
     }
```

Reply:
Patternmatching is usually much more readable then `isInstanceOf` and `asInstanceOf`
`Option.fold` is nicer then `Option.map().getOrElse()`

Reply:
Changed if elses to patternmatching - It's really much more readable